audio_fifo.go

```diff
@@ -0,0 +1,52 @@
+package astiav
+
+//#include <libavutil/audio_fifo.h>
+//#include <stdlib.h>
+import "C"
+import "unsafe"
+
+// https://github.com/FFmpeg/FFmpeg/blob/n7.0/libavutil/audio_fifo.c#L37
+type AudioFifo struct {
+	c *C.AVAudioFifo
+}
+
+func newAudioFifoFromC(c *C.struct_AVAudioFifo) *AudioFifo {
+	if c == nil {
+		return nil
+	}
+	return &AudioFifo{c: c}
+}
+
+func AllocAudioFifo(sampleFmt SampleFormat, channels int, nbSamples int) *AudioFifo {
+	return newAudioFifoFromC(C.av_audio_fifo_alloc(C.enum_AVSampleFormat(sampleFmt), C.int(channels), C.int(nbSamples)))
+}
+
+func (a *AudioFifo) Realloc(nbSamples int) error {
+	return newError(C.av_audio_fifo_realloc(a.c, C.int(nbSamples)))
+}
+
+func (a *AudioFifo) Size() int {
+	return int(C.av_audio_fifo_size(a.c))
+}
+
+func (a *AudioFifo) Space() int {
+	return int(C.av_audio_fifo_space(a.c))
+}
+
+func (a *AudioFifo) Write(f *Frame) (int, error) {
+	frameRawData := (**uint8)(unsafe.Pointer(&f.c.data[0]))
+	ret := int(C.av_audio_fifo_write(a.c, (*unsafe.Pointer)(unsafe.Pointer(frameRawData)), C.int(f.NbSamples())))
+	if ret < 0 {
+		return ret, newError(C.int(ret))
+	}
+	return ret, nil
+}
+
+func (a *AudioFifo) Read(f *Frame) int {
+	frameRawData := (**uint8)(unsafe.Pointer(&f.c.data[0]))
```

Comment:
Can you avoid creating this variable and use `(*unsafe.Pointer)(unsafe.Pointer(&f.c.data[0]))` instead of `(*unsafe.Pointer)(unsafe.Pointer(frameRawData))`?